ServiceBus/ServiceBus/ProductService.svc.cs

```diff
@@ -136,63 +141,74 @@ public Result AddProduct(string name, double price, Guid guid)
         /// <param name="guid">ID of a product</param>
         /// <param name="name">New name for a product</param>
         /// <param name="price">New price for a product</param>
+        /// <param name="pType">Type of a product</param>
         /// <returns>Modified product</returns>
-        public SharedLibs.DataContracts.Product EditProduct(Guid guid, string name, double price)
+        public SharedLibs.DataContracts.Product EditProduct(Guid guid, string name, double price, ProductType pType = null, bool enabled = true, bool headliner = false)
         {
             try
             {
-                var originalProduct = GetProduct(guid);
-
-                if (originalProduct.Result.ResultType == ResultType.Success)
-                {
-                    var editableProduct = new Product()
-                    {
-                        Id = originalProduct.ID,
-                        Name = originalProduct.Name,
-                        Price = originalProduct.Price
-                    };
-
                     using (var context = new ServiceBusDatabaseEntities())
                     {
-                        context.Products.Attach(editableProduct);

```

Comment:
It works when value types of the objects are changed. Referential are ignored. 

I chosen this way because I wanted to distinguish a scenario when someone want to change something but input parameters are exactly the same as those which are already present in db. I can try something like 

bool flag = false;

if(att is valid){ ... ; flag = true;}
if(att2 is valid) {... ; flag = true;}
...
if (flag){
saveChanges(); 
}
else{ Result("nothing has been changed") }